File: Day 1 2020/day12020.js
// js version of py functions for practice
// final working solutions

// DAY 1
// Part 1

// <first function>
function findPair(nums) {
    for (let num of nums) {
      for (let numo of nums) {
        let newVar = num + numo;
        if (newVar == 2020) {
          console.log(`${num} multiplied by ${newVar} equals ${num * newVar}`);
        }
      }
    }
  };


// <second function that's better>
function findPair(nums) {
    for (const num of nums) {
      let sub = 2020 - num;
       if (nums.includes(sub)) 
        console.log(`${num} multiplied by ${sub} equals ${num * sub}`);
      }
  };


// Part 2

function threeAdds(nums) {
    for (const num of nums) {
      for (const item of nums){
        const small = 2020 - num - item
        if (nums.includes(small)){
          console.log(`${num} multiplied by ${item} and ${small} equals ${num * item * small}`);
    }}}
  };

File: Day 1 2020/day_1_2020.py
# ADVENT OF CODE 2020 #
# Final working solutions

# DAY 1
# Part 1

# <First way I solved the problem> 

def find_pair(nums):
    for num in nums:
        for numo in nums:
            new_var = num + numo
            if new_var == 2020:
                return print(f"{num} multiplied by {numo} equals {num * numo}!")

# <Second way I solved it is better bc runtime>

def find_pair(nums):           
    for num in nums:
        sub = 2020 - num
        if sub in nums:
            return print(f"{num} multiplied by {sub} equals {num * sub}!")


# Part 2

def three_adds(nums):
    for num in nums:
        for item in nums:
            small = 2020 - num - item
            if small in nums:
                return print(f"{num} multiplied by {item} and {small} equals {num * item * small}!")

